src/main/java/org/takes/rq/RqLive.java

```diff
@@ -95,29 +97,16 @@ private static Request parse(final InputStream input) throws IOException {
                     break;
                 }
                 data = new Opt.Single<Integer>(input.read());
-                if (data.get() != ' ' && data.get() != '\t') {
-                    head.add(new String(baos.toByteArray()));
-                    baos.reset();
-                }
+                addHeader(data, baos, head);
```

Comment:
@yegor256 head is final and can't be reassigned, but I'm still not convinced that `withHeader` should return the data.